### ISSUE TITLE
[Enhancement] Research points rebalance

### ISSUE THREAD

**Issue by Fjolsvin** [Azanor/thaumcraft-5#102]
I like the new research system, with no need of exploring tons of nodes for research points. However, it is not challenging at all. To connect all the aspects is a complete no-brainer with huge surplus of research points. An advanced research (4 hex side) can be easily done with 50-70 research points left, i. e. a player can do up to 70 wrong moves and still be able to finish a research. Moreover "Research Expertise" and "Research Mastery" both has close-to-zero value, as I really don't need chance to regain my research point nor aditional primal aspects for knowledge fragments, when I already have that huge surplus of them. And finally, I do like the idea to get randomly generated amount of research points for a research note, however I really don't care, if I have 15 or 17 aer research points, when I need 2 or 3 to complete the research.

I don't say it has to be a hardcore puzzle (notwithstanding I'd appreciate that), but 70 mistakes tolerance is far too much. I'd suggest to significantly reduce the amount of research points, to the very edge of doability. Also, I'm not sure if "practical knowledge" of an aspect (gain from exploration with Thaumometer) has any effect, but it could permanently increase the amount of research point for the certain aspect; in this case I strongly suggest to move the initial amount of research points under the edge of doability, so the research can be finished only with "practical knowledge" or knowledge from fragments.

E. g.: Sitting at research table, I need lets say 3 aer research points to be able to complete a research, but a research note always generates only 0-1 aer research points. I can try to find another solution, typically more complex, nevertheless with lower consumption of aer. Or I can use knowledge fragment. Or I can go exploring, get a "practical knowledge" of aer and/or any compound aspect containing aer and with that knowledge make a new research note, now with more research points.

EDIT: Or much more gaps to the hex sheet, to force a player to combine higher aspects, thus use more primal ones.

**Comment by Xirema** [comment]
I'd say this is partially a result of simplifying the aspects, which means that you never need to go more than three levels of aspect combining to get access any given aspect.

I wonder if maybe we need a penalty for using the same aspect(s) over and over? Like every time you use an aspect in the note, the next costs more, meaning for particularly complex notes, you'd be forced to use alternate aspects to complete it.

**Comment by Fjolsvin** [comment]
The simplifying of the aspects is not a cause. You don't need to use compound aspects of higher tiers at all. There are only 5 aspects you need to connect to 1st tier compound aspect, all the other aspects can be connected to primary aspect directly. I've never used 2nd or 3rd tier aspect in my research. Therefore it actually doesn't matter how many tiers there are.

The penalty idea doesn't solve the problem with surpluses of aspects. I actually did the same - just to have the research notes more colorful - and I still finished all the researches with huge number of aspects unused and I never ever run out of a single primal aspect.

Building on your idea, penalty for using every additional aspect would be even better, as you'd be forced to use the shortest connections possible, thus forced to combine higher tier aspects. Actually it's an alternative to adding more gaps to the hex sheet, both make you use more aspects. Nevertheless, why to make the research more costly, if the number of aspects per research note can be lowered simply?

**Comment by Xirema** [comment]
Well, my strategy has been to find a relatively easy to link primal aspect, link everything to that, and then join them in the middle. It's efficient, but not terribly challenging and it leaves tons of unused points. Penalties for using the same aspect over&over would encourage a system where many varied aspects (primal and compound) are used to combine everything.

Also, the reason I called attention to the simplified compounds is because higher tier "starting points" would help fix this as well, but because the highest tier is 3-ish, it's a minor impact.

**Comment by Fjolsvin** [comment]
Trying to improve research mechanics, following up current issues I mentioned above, I evolved the idea little more:

1) A player has 1-2 (random) research point of every primal aspect available from the begining (as he read the Thaumonomicon already). E.g. Steve draws his very first research note - Nitor. With the pool of aspects he has, he has a hard time to join Lux and Ignis though. He finds no way to connect those aspects. For now.

2) Exploring an environment with thaumometer, a player can gain one permanent primal or compound aspect every first time, he discovers one, e.g.: Steve grabs his new-made Thaumometer and explores a cow. He gains +1 terra, +1 bestia. Now when he explores a grass block, he won't get another terra, as he already has the practical knowledge of the aspect, but he gains +1 herba.

Back to Nitor research, Steve explored a torch and gained +1 lux. Clever idea. He still can't use "no-brainer" solution (lux)-ignis-lux-ignis-lux-(ignis), but if he was lucky enough to gain 2 aer and 2 ignis from the begining, together with the newly discovered lux, he can try (lux)-ignis-lux-aer-lux-(ignis).

If he wasn't lucky enough and have only 1 ignis, but 2 perditio, he still can find another solution. By exploring an empty bottle, he gains vacuos and he can make more imaginative way (lux)-aer-vacuos-perditio-gellum-(ignis), costing 2 aer, 1 vacuos (bottle), 2 perditio and only 1 ignis. Of course, there are another ways to connect the aspects.

3) Thus, after a player finds all the compound aspects in a nature, his research pool will be 2-3 research points of every primal aspect and 1 research point of every compound aspects, resulting Research mastery shift-click advantage has its sense again, as there are compound aspects in the pool that can be shift-clicked. Depleting of any aspect should result to indicating 0 aspects instead of removal of the aspect, so it can be shift-clicked again.

4) The pool mentioned above should be enough to do all basic researches (3-side-hex) easily. Advanced researches (4-side-hex) and especialy expert researches (5-side-hex) still needs more primal aspects though. And now the lost knowledge fragments come to the scene. It can stay as it is, giving +1 to every primal aspect for a particular research or, as the lost knowledge fragments are quite rare, giving few random primal aspects permanently, e.g. +1 ignis and +1 aer.

5) The last source of research points could be nodes agian. Unlike other objects, by exploring more nodes a player would gain +1 research point of the aspect a node is emiting. That gives a player a reasonable choice: He can think more about a research to find an imaginative solution with research points he already has or he can choose a "no-brainer" way and explore tons of nodes outside. Some nodes are not reachable too easy, however if one feels there is too high density of nodes for this mechanics, it can be altered by only a certain chance to get research point by examinig a node.

Note: I realized I'm combining research-points-per-research-note and general-research-points-pool mechanics. I do hope it's technically possible. Here is a summary:

a) For every research note a player gains
1-2 (random number) research points of every primal aspect (3-side-hex note) or
2-3 (random number) research points of every primal aspect (4-side-hex note) or
3-4 (random number) research points of every primal aspect (5-side-hex note).
b) For every aspect discovered by Thaumometer for the first time in the nature (practical knowledge) a player gains
+1 research point (primal or compound) of the corresponding aspect, additional to letter a), if the note was drawn after the discovery.
c) For every knowledge fragment used a player gains
+1 research point of 1-3 (random) primal aspect(s) , additional to a), if the note was drawn after the discovery.
d) For every node explored, a player has certain % chance to gain
+1 research point (primal or compound) of the corresponding aspect, additional to letter a), if the note was drawn after the node was explored.

Alternatives:
c) Research notes can be used the same way as now instead, i. e. +1 research point of every primal aspect for a singe research note only.
d) Chance to gain additional research points by exploring nodes can be altered to any number from 0 to 100 %, when 0 is kinda hardcore (and the reasonable doability of some researches need to be tested) and 100 is kinda foolproof, making the exploration a way to amass a huge number of research points once more.

**Comment by Xirema** [comment]
I'd suggest fractions of a research point, i.e. a 50 strength node yields 50/100 of a research point. When they get a whole research point, then it gets added to their [permanent] pool. Maybe similar mechanics to what I described before: first point obly requires 100, second requires 200, third requires 400, fourth requires 800... Basically, you gain benefits from finding lots of nodes, but exponential growth keeps you from just spamming research of nodes to fill your pool.

**Comment by Fjolsvin** [comment]
Great idea. I tried to hold it as simple as possible, but I like the idea.

However, I did some additional calculations. There are lots of exceptions, but roughly a player needs to use up to 6 research points (RP) to finish an easy research, 10 RP for an advanced research and 14 RP for an expert research. Also the shorter a distance between aspects to connect, the bigger consumption of primal aspect RP per an aspect used, reaching a value from 1,7 to 2,2. Thus a player needs to use roughly 10-13 primal RP to finish an easy research, 17-22 primal RP for advanced research and 24-31 primal RP for expert research.

By giving a player numbers I suggested before (i. e. 3-4 research points of every primal aspect for 5-side-hex note etc.), he'd gain 6-12 primal RP, where he needs 10-13 (ok), but 12-18 primal RP, where he needs 17-22 and 18-24 RP, where he needs 24-31. Together with compound RP gain by exploration and possibly research notes, it is too much again.

Therefore I had to modify some ideas mentioned above:

- Do not increase the number of initial RP for advanced researches, i. e. a player gains only 1-2 (random number) research points of every primal aspect for any note (regardless of size of the hex grid).
- Keep the usage of Knowledge fragments as it is, i. e. bonus for one research note only.
- Cancel the idea about RP for aura nodes completely or if so, lower the chance significantly, e. g. 50 aer node yields 50/1000 (5 %) chance to get another permanent primal RP and if so, the next one 1/2000, the 3rd only 1/4000 etc.

Thus, after finishing an exploration, a player will have 2-3 RP of every primal plus 1 RP for every compound aspect, making all the small research notes really easy, medium research notes still quite easy and big research notes reasonably difficult (not that much though).

(By now, the modifications were changed in the original post)
